### ISSUE TITLE
3.1 broke -CC (iTerm2 integration)

### ISSUE THREAD

**Issue by fishy** [tmux/tmux#2182]
### Issue description

See also the bug report on iTerm2: https://gitlab.com/gnachman/iterm2/-/issues/8850

Steps to reproduce, in iTerm2:

- `tmux -CC`
- hit `ESC` to detach

Instead of detaching cleanly, it says "Detaching" but hang there, also the iterm windows are no longer responsive.

Reproducible on latest master (f3d6d4e8).

### Required information

Please provide the following information:

* tmux version (`tmux -V`).
3.1
* Platform (`uname -sp`).
Darwin i386
* $TERM inside and outside of tmux (`echo $TERM`).
xterm-256color
* Logs from tmux (`tmux kill-server; tmux -vv new`).
[tmux-client-77690.log](https://github.com/tmux/tmux/files/4536936/tmux-client-77690.log)
[tmux-server-77692.log](https://github.com/tmux/tmux/files/4536937/tmux-server-77692.log)

**Comment by nicm** [comment]
I see this too, but I don't know what iTerm2 is waiting for here. The client has definitely exited when you do detach, and in control mode the file descriptor is not held open by the server:

~~~~
$ pgrep -lf tmux
25004 ./tmux -vv -CC
25051 ./tmux -vv -CC
<press Escape in the other window>
$ pgrep -lf tmux
25004 ./tmux -vv -CC
~~~~

@gnachman?

**Comment by nicm** [comment]
I wonder if the %exit is going missing.

**Comment by nicm** [comment]
Yes, we are incorrectly closing the `stdout` fd, looks like a typo. This seems to fix it: [x.diff.txt](https://github.com/tmux/tmux/files/4538304/x.diff.txt)
~~~~
diff --git a/client.c b/client.c
index 108873e7..4e7f7dcc 100644
--- a/client.c
+++ b/client.c
@@ -520,7 +520,7 @@ client_write_open(void *data, size_t datalen)
 			errno = EBADF;
 		else {
 			cf->fd = dup(msg->fd);
-			if (client_flags & CLIENT_CONTROL)
+			if (~client_flags & CLIENT_CONTROL)
 				close(msg->fd); /* can only be used once */
 		}
 	}
@@ -675,7 +675,8 @@ client_read_open(void *data, size_t datalen)
 			errno = EBADF;
 		else {
 			cf->fd = dup(msg->fd);
-			close(msg->fd); /* can only be used once */
+			if (~client_flags & CLIENT_CONTROL)
+				close(msg->fd); /* can only be used once */
 		}
 	}
 	if (cf->fd == -1) {

**Comment by gnachman** [comment]
I can confirm the fix works.

Not sure what happened…I tested the last RC pretty well, but maybe I somehow didn't test detaching.

**Comment by nicm** [comment]
OK, well, no worries I will do a patch release in a few days.

**Comment by nicm** [comment]
Thanks :-).

**Comment by nicm** [comment]
For the moment you can build from the `3.1a-rc` branch.

**Comment by tommyjcarpenter** [comment]
Tmux 3.1, freshly installed with brew for me this morning on OSX in Iterm2, just immediately exits:

```
 ~  tmux                                                                                                                                                                                                                                                                                    
[exited]
^[[?62;4c⏎                                                                                                                                                                                                                                                                                                                     ~  [?62;4c
```

**Comment by und3fined** [comment]
@nicm please make a tarball assets for release

**Comment by nicm** [comment]
I have released 3.1a with this fix now.
